### ISSUE TITLE
Change Request: Dissable errors for directories with files that are all unsupported.

### ISSUE THREAD

**Issue by Reboot-Codes** [eslint/eslint#16003]
### ESLint version

v8.5.0

### What problem do you want to solve?

Currently there are some directories which are empty, or contain files that are not supported by ESLint (e.g. different language than JS/TS); when ESLint encounters one of these directories, it throws an error and exits with a non-zero exit code. I would expect that the `--no-error-on-unmatched-pattern` CLI option would disable this or adding that pattern to `.eslintignore`, it doesn't. **(Slightly Important) Context:** This is a mono-repo with many different projects in the `repo-root/packages/` directory. I would like all JS/TS files in these to be linted (after being formatted by Prettier, which is working as expected) despite them containing other types of files (like `.rs`, `.go`, `.py`, etc) which aren't supported by ESLint (as far as I know) and they have their own linters.

### What do you think is the correct solution?

I get that this might be an issue for some, but I'd like a way to turn this specific error off; either through a CLI switch, or a configuration option in `.eslintrc.*`. Or just make ESLint a bit more "lenient" like Prettier (e.g. doesn't complain about unsupported files).

### Participation

- [X] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Hi @Reboot-Codes, thanks for the issue!

> Currently there are some directories which are empty, or contain files that are not supported by ESLint (e.g. different language than JS/TS); when ESLint encounters one of these directories, it throws an error and exits with a non-zero exit code. I would expect that the --no-error-on-unmatched-pattern CLI option would disable this or adding that pattern to .eslintignore, it doesn't. 

How did you run ESLint, and what error did you get? `--no-error-on-unmatched-pattern` should prevent errors for unmatched patterns and patterns that match only ignored files.

**Comment by Reboot-Codes** [comment]
```
[error] No supported files were found in the directory: "./packages/landing/public".
```

**Context:** A Next.js project with an empty public directory.

**Comment by mdjermanovic** [comment]
This seems to be an [error message](https://github.com/prettier/prettier/blob/5530ad24b952e0c48d6b13a17fbba0c45b645f8d/src/cli/expand-patterns.js#L153) from Prettier, I don't think we have this error message in ESLint.

**Comment by Reboot-Codes** [comment]
Weird, well, looks like I'll have to go fix prettier then, thank you for your help!
